.github/actions/cherry-pick/action.yml

```
@@ -0,0 +1,206 @@
+
+        # Case 2: Label was added
+        if [ "${{ github.event.action }}" = "labeled" ]; then
+          LABEL_NAME="${{ github.event.label.name }}"
+          # Check if it's a backport label
+          if [[ "$LABEL_NAME" =~ ^backport-(.+)$ ]]; then
+            # Check if PR is already merged
+            if [ "${{ github.event.pull_request.merged }}" = "true" ]; then
+              echo "should_run=true" >> $GITHUB_OUTPUT
+              echo "reason=label_added_to_merged_pr" >> $GITHUB_OUTPUT
+              exit 0
+            else
+              echo "should_run=false" >> $GITHUB_OUTPUT
+              echo "reason=label_added_to_open_pr" >> $GITHUB_OUTPUT
+              echo "Backport label added to open PR. Will run after PR is merged."
+              exit 0
+            fi
+          else
+            echo "should_run=false" >> $GITHUB_OUTPUT
+            echo "reason=non_backport_label" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+        fi
+
+        echo "should_run=false" >> $GITHUB_OUTPUT
+        echo "reason=unknown" >> $GITHUB_OUTPUT
+    - name: Configure Git
+      if: steps.should_run.outputs.should_run == 'true'
+      shell: bash
+      run: |
+        git config --global user.name "authentik-automation[bot]"
+        git config --global user.email "135050075+authentik-automation[bot]@users.noreply.github.com"
+    - name: Extract backport labels and cherry-pick
+      if: steps.should_run.outputs.should_run == 'true'
+      shell: bash
+      env:
+        GITHUB_TOKEN: ${{ inputs.token }}
+      run: |
+        # Get PR number and commit SHA
+        PR_NUMBER="${{ github.event.pull_request.number }}"
+        COMMIT_SHA="${{ github.event.pull_request.merge_commit_sha }}"
+        PR_TITLE="${{ github.event.pull_request.title }}"
+        PR_AUTHOR="${{ github.event.pull_request.user.login }}"
+
+        echo "Processing PR #$PR_NUMBER (reason: ${{ steps.should_run.outputs.reason }})"
+
+        # Determine which labels to process
+        if [ "${{ steps.should_run.outputs.reason }}" = "label_added_to_merged_pr" ]; then
+          # Only process the specific label that was just added
+          LABEL_NAME="${{ github.event.label.name }}"
+          if [[ "$LABEL_NAME" =~ ^backport-(.+)$ ]]; then
+            LABELS="$LABEL_NAME"
+          else
+            echo "Label $LABEL_NAME does not match backport pattern"
+            exit 0
+          fi
+        else
+          # PR was just merged, process all backport labels
+          LABELS=$(gh pr view $PR_NUMBER --json labels --jq '.labels[].name' | grep '^backport-' || true)
+        fi
+
+          if [ -z "$LABELS" ]; then
+            echo "No backport labels found"
+            exit 0
+          fi
+
+          echo "Found backport labels: $LABELS"
+
+          # Process each backport label
+          echo "$LABELS" | while read -r label; do
+            if [[ "$label" =~ ^backport-(.+)$ ]]; then
+              TARGET_BRANCH="${BASH_REMATCH[1]}"
```

```suggestion
            if [[ "$label" =~ ^backport/(.+)$ ]]; then
              TARGET_BRANCH="${BASH_REMATCH[1]}"
```
with the other change